trace/trace.go

```diff
@@ -0,0 +1,146 @@
+package trace
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"regexp"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+const (
+	maxVersion       = 254
+	supportedVersion = 0
+
+	TraceparentHeader = "traceparent"
+	TracestateHeader  = "tracestate"
+)
+
+// source from opentelmetry-go, https://github.com/open-telemetry/opentelemetry-go/blob/main/propagation/trace_context.go#L44.
+var traceCtxRegExp = regexp.MustCompile("^(?P<version>[0-9a-f]{2})-(?P<traceID>[a-f0-9]{32})-(?P<spanID>[a-f0-9]{16})-(?P<traceFlags>[a-f0-9]{2})(?:-.*)?$")
+
+// Traceparent gets traceparent from spancontext.
+func Traceparent(sc trace.SpanContext) string {
+	flags := sc.TraceFlags() & trace.FlagsSampled
+
+	return fmt.Sprintf("%.2x-%s-%s-%s",
+		supportedVersion,
+		sc.TraceID(),
+		sc.SpanID(),
+		flags)
+}
+
+// ID gets traceid from context.
+func ID(ctx context.Context) string {
+	sc := trace.SpanContextFromContext(ctx)
+	if sc.HasTraceID() {
+		return sc.TraceID().String()
+	}
+	return ""
+}
+
+// NewSpanContextFromTrace generates span context.
+func NewSpanContextFromTrace(traceparent, tracestate string) trace.SpanContext {
+	sc := SpanContextFromW3CString(traceparent)
+	ts := StateFromW3CString(tracestate)
+
+	return sc.WithTraceState(ts)
+}
+
+// w3c reference: https://www.w3.org/TR/trace-context/#traceparent-header.
+// traceparent: 00-0af7651916cd43dd8448eb211c80319c-b7ad6b7169203331-01.
+// SpanContextFromW3CString generates span context by traceparent.
+func SpanContextFromW3CString(traceparent string) trace.SpanContext {
```

Comment:
Please use the method we're currently using in dapr/dapr: https://github.dev/dapr/dapr/blob/a64a8a97c59c0678bec4d754b21d837d25a38f4c/pkg/diagnostics/tracing.go#L107-L108